tests/conftest.py

```diff
@@ -184,29 +194,171 @@ def is_provider_installed(cls) -> bool:
 
 
 @pytest.fixture
-def create_config(tmp_path):
-    """Helper to create a config file in disk.
+def create_charmcraft_yaml(tmp_path: pathlib.Path):
+    """Helper to create a charmcraft.yaml file in disk.
 
     If content is not given, create a minimum valid file.
     """
 
-    def create_config(text=None):
-        if text is None:
-            text = """
+    def create_charmcraft_yaml(content="LEGACY", content_extend=None):
+        if content is None:
+            charmcraft_yaml = ""
+        elif content == "LEGACY":
+            charmcraft_yaml = dedent(
+                """\
                 type: charm
-                bases:
-                  - build-on:
-                    - name: test-build-name
-                      channel: test-build-channel
-                    run-on:
-                    - name: test-build-name
-                      channel: test-build-channel
-            """
-        test_file = tmp_path / "charmcraft.yaml"
-        test_file.write_text(text)
+                """
+            )
+        else:
+            charmcraft_yaml = content
+
+        if content_extend is not None:
+            charmcraft_yaml += "\n" + content_extend
+
+        if charmcraft_yaml is not None:
+            test_config_file = tmp_path / "charmcraft.yaml"
+            test_config_file.write_text(charmcraft_yaml)
+        else:
+            try:
+                os.remove(tmp_path / "charmcraft.yaml")
+            except OSError:
+                pass
+
+        return tmp_path
+
+    return create_charmcraft_yaml
+
+
+@pytest.fixture
+def create_metadata_yaml(tmp_path: pathlib.Path):
+    """Helper to create a metadata.yaml file in disk.
+
+    If content is not given, create a minimum valid file.
+    """
+
+    def create_metadata_yaml(content="LEGACY", content_extend=None):
```

Comment:
content="LEGACY" makes a little noise for me, 

can we make the default content value None or "" and have a different bool like create_metadata_file=False

@lengau ?

Reply:
I was plan to add the `content="COMBINED"` for the merged `charmcraft.yaml`.
I am not sure the best practices here, since we need 2 default values, and a way to not create / delete it.

Reply:
I agree. The more flexible we're trying to make this the more we seem to be building business logic into the test setups.

I think the amount of logic that was in the original is about the maximum we want to do: either write the full file or have a sentinel value to write a default file. Having content and a content extension can make a failing test harder to reason about. The more locally we can keep things scoped, the better, even if it means we're repeating the name/summary/description lines a dozen times.

I also keep coming back to the thought of putting the "charm bases" in their own data directories. While charmcraft doesn't currently distinguish unit tests from integration tests, these feel pretty integration-y to me. Perhaps a better helper to have would be a `prepare_config` that receives a directory containing the relevant YAML files and just copies the contents of that directory into `tmp_dir`?

Reply:
Well, in that case we may just duplicate the full yaml files without reuse any common parts. That will make less business logic, but a lot more yaml lines in the test cases.

I still don't think we need to copy the contents of an directory, it usually just 1 or 2 files.

Reply:
yes, this type of logic makes it really hard on future maintenance